exla/lib/exla/defn.ex

```diff
@@ -307,7 +307,7 @@ defmodule EXLA.Defn do
     {res, cache} = recur_flatten(expr, state, new_cache(outfeed))
     outfeed = cache |> get_outfeed() |> Outfeed.close(function)
 
-    Value.return(function, res)
+    Value.func_return(function, res)
```

Comment:
We should also change `to_root_computation` and here: https://github.com/elixir-nx/nx/blob/c7cdd847a0813897d6ddb7ba7b315a5ff8e73f1a/exla/test/support/exla_helpers.ex#L18

Reply:
Updated!

Reply:
Yeah, I changed both of these places. The errors were because we actually created the MLIR function with wrong return type in those cases, and it seems that using `stablehlo.return` omitted a validation that `func.return` does trigger :D